extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js

```diff
@@ -107,15 +109,35 @@ export class AmpAdNetworkAdsenseImpl extends AmpA4A {
      */
     this.ampAnalyticsConfig_ = null;
 
+    /**
+     * amp-analytics element generated based on this.ampAnalyticsConfig_
+     * @type {?Element}
+     * @private
+     */
+    this.ampAnalyticsElement_ = null;
+
+    /**
+     * Config to generate amp-analytics element for reporting on page load
+     * metrics
+     * @type {?JsonObject}
+     * @private
+     */
+    this.ampAnalyticsPageLoadMetricsConfig_ = null;
```

Comment:
Since we're interested in page-level metrics here (when the full page becomes visible, the first viewport ready for the page, etc.), we really only want one config and one analytics element for the page.  So you should probably attach these to `win` somewhere.

Reply:
Ok. So is amp-ad-network-adsense-impl.js the wrong place for this to be located?

Reply:
Probably.  It feels like it belongs with the other CSI monitoring code, which is in ads/google/a4a, and is shared between type=adsense and type=doubleclick.  That way you get both with one change.

Reply:
Ok, I'll move it there. Thanks.